### ISSUE TITLE
Change behaviour of ELASTIC_APM_INSTRUMENT config

### ISSUE THREAD

**Issue by alvarolobato** [elastic/apm-agent-ruby#467]
Add `ELASTIC_APM_INSTRUMENT` for disabling instrumentation modules. This should not affect OpenTracing or other custom instrumentation.

See https://github.com/elastic/apm/issues/49

---

**Update @mikker**: Currently blocked by https://github.com/elastic/apm/issues/92 which will probably move this and a few other things around.

**Comment by mikker** [comment]
We already have an option called `instrument` but it means something else. Right now it switches all transactions on/off (and thereby also spans). Keeping error tracking.

Changing the behaviour of this would be a breaking change. If users want the functionality to match the new definition of this, they can disable most auto-instrumentation with `config.disabled_instrumentations =config.available_instrumentations`.

If the option is to be available via CentralConfig we also need to change the spies. Disabling one right now means not loading its _Spy_, ie. file. Every spy would need to have a hot-swappable `disabled` flag like for example [the net/http one has](https://github.com/elastic/apm-agent-ruby/blob/master/lib/elastic_apm/spies/net_http.rb#L12-L29). That way we'll load all files, then disable/enable the ones the user has chosen.

We also don't have a way to distinguish between manually started transactions and spans and agent-defined ones. I suppose we'd need to add some sort of flag to the `start_X` methods to distinguish, eg. `ElasticAPM.start_span 'test', internal: true` or `auto: true` or `from_agent: true`. All of them feel a bit weird but I do not want to put the responsibility of adding a flag on the user.

@estolfo, @axw _wdyt?_

**Comment by estolfo** [comment]
How does this relate to https://github.com/elastic/apm/issues/92 ? It is a completely separate issue?

I would also bring up these important points in the agent issue, as Thomas said something similar to (4) regarding the node agent.

**Comment by mikker** [comment]
This issue was created before that discussion but you're right that it's deeply affected by it. I think we should sit on this and maybe resolve it along with that other issue instead which is still (sigh) being debated. Agree?

**Comment by estolfo** [comment]
Yes, this seems related to the other issue and they should be resolved/decided together.

**Comment by mikker** [comment]
Closing as `instrument` will be replaced by `recording`.
